weaviate/collections/batch/base.py

```diff
@@ -233,9 +231,21 @@ def __init__(
         # do 62 secs to give us some buffer to the "per-minute" calculation
         self.__fix_rate_batching_base_time = 62
 
+        self.__loop.run_until_complete(self.__make_asyncio_locks)
+
         self.__bg_thread = self.__start_bg_threads()
         self.__bg_thread_exception: Optional[Exception] = None
 
+    async def __make_asyncio_locks(self) -> None:
+        """Create the locks in the context of the running event loop so that internal `asyncio.get_event_loop()` calls work."""
+        self.__active_requests_lock = asyncio.Lock()
+        self.__uuid_lookup_lock = asyncio.Lock()
+        self.__results_lock = asyncio.Lock()
+
+    async def __release_asyncio_lock(self, lock: asyncio.Lock) -> None:
+        """Release the lock in the context of the running event loop so that internal `asyncio.get_event_loop()` calls work."""
+        return lock.release()
```

Comment:
I remember that you said that when you replaced the normal locks with async locks there were other problems

- what were those problems?
- are they mitigated somehow?

Reply:
- The problems were with specific timings in the batching logic
  - Due to the previous blocking, the batching thread would be faster at looping than the event loop thread would be at returning
  - Now, it's the other way around leading to small batches being sent (~50-100 objs per batch) which slows the whole process down
- The mitigation to this is the above introduced logic, which you commented on, that sleeps until it there are a recommended number of objects and then sends that batch rather than being optimistic by pulling less than or equal that many